yb-voyager/cmd/analyzeSchema.go

```diff
@@ -686,9 +688,97 @@ var UnsupportedIndexDatatypes = []string{
+	schemaName := alterTableNode.AlterTableStmt.Relation.Schemaname
+	tableName := alterTableNode.AlterTableStmt.Relation.Relname
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	alterCmd := alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd()
+	unsupportedColumnsForTable, ok := columnsWithUnsupportedIndexDatatypes[fullyQualifiedName]
+	if !ok {
+		return
+	}
+	if alterCmd.GetSubtype() != pg_query.AlterTableType_AT_AddConstraint {
+		return
+	}
+	if !slices.Contains([]pg_query.ConstrType{pg_query.ConstrType_CONSTR_PRIMARY, pg_query.ConstrType_CONSTR_UNIQUE},
+		alterCmd.GetDef().GetConstraint().GetContype()) {
+		return
+	}
+	columns := alterCmd.GetDef().GetConstraint().GetKeys()
+	for _, col := range columns {
+		colName := col.GetString_().Sval
+		typeName, ok := unsupportedColumnsForTable[colName]
+		if ok {
+			displayName := fmt.Sprintf("%s, constraint: %s", fullyQualifiedName, alterCmd.GetDef().GetConstraint().GetConname())
+			reportCase(fpath, fmt.Sprintf(ISSUE_CONSTRAINT_WITH_COMPLEX_DATATYPES, typeName), "https://github.com/yugabyte/yugabyte-db/issues/9698",
+				"Refer to the docs link for the workaround", "TABLE", displayName, sqlStmtInfo.formattedStmt,
+				UNSUPPORTED_FEATURES, PK_UK_CONSTRAINT_ON_UNSUPPORTED_TYPE)
+			return
+		}
+	}
+}
+
+func reportUnsupportedConstraintsOnComplexDatatypesInCreate(createTableNode *pg_query.Node_CreateStmt, sqlStmtInfo sqlInfo, fpath string) {
+	schemaName := createTableNode.CreateStmt.Relation.Schemaname
+	tableName := createTableNode.CreateStmt.Relation.Relname
+	columns := createTableNode.CreateStmt.TableElts
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	unsupportedColumnsForTable, ok := columnsWithUnsupportedIndexDatatypes[fullyQualifiedName]
+	if !ok {
+		return
+	}
+	reportConstraintIfrequired := func(con *pg_query.Constraint, colNames []string, typeName string) {
+		conType := con.GetContype()
+		generatedConName := generateConstraintName(conType, fullyQualifiedName, colNames)
+		specifiedConstraintName := con.GetConname()
+		conName := lo.Ternary(specifiedConstraintName == "", generatedConName, specifiedConstraintName)
+		//report the PK / Unique constraint in CREATE TABLE on this column
+		if slices.Contains([]pg_query.ConstrType{pg_query.ConstrType_CONSTR_PRIMARY, pg_query.ConstrType_CONSTR_UNIQUE}, conType) {
+			summaryMap["TABLE"].invalidCount[fullyQualifiedName] = true
+			reportCase(fpath, fmt.Sprintf(ISSUE_CONSTRAINT_WITH_COMPLEX_DATATYPES, typeName), "https://github.com/yugabyte/yugabyte-db/issues/9698",
+				"Refer to the docs link for the workaround", "TABLE", fmt.Sprintf("%s, constraint: %s", fullyQualifiedName, conName), sqlStmtInfo.formattedStmt,
+				UNSUPPORTED_FEATURES, PK_UK_CONSTRAINT_ON_UNSUPPORTED_TYPE)
+		}
+	}
+	for _, column := range columns {
+		if column.GetColumnDef() != nil {
```

Comment:
just add comments in both the cases to explain again that in some cases constraint can be defined along with column, and in some cases, constraint is a defined separately.